servicetalk-grpc-protoc/src/main/java/io/servicetalk/grpc/protoc/Generator.java

```diff
@@ -415,64 +452,66 @@ private void addServiceFactory(final State state, final TypeSpec.Builder service
                 .addMethod(registerRoutesMethodSpecBuilder.build())
                 .build();
 
-        final TypeSpec.Builder serviceFactoryClassSpecBuilder = classBuilder(serviceFactoryClass)
+        final TypeSpec.Builder serviceFactoryClassSpecBuilder = classBuilder(state.serviceFactoryClass)
                 .addModifiers(PUBLIC, STATIC, FINAL)
                 .superclass(ParameterizedTypeName.get(GrpcServiceFactory, state.serviceFilterClass, state.serviceClass,
                         state.serviceFilterFactoryClass))
+                // Add ServiceFactory constructors for blocking and async services with and without content codings and
+                // execution strategy
                 .addMethod(constructorBuilder()
                         .addModifiers(PUBLIC)
                         .addParameter(state.serviceClass, service, FINAL)
-                        .addStatement("super(new $T().$L)", builderClass,
+                        .addStatement("this(new $T().$L)", builderClass,
```

Comment:
not related to this PR: why do we need a `ServiceFactory` with all the constructor variants and a `Builder`? Seems redundant to have both patterns.

/cc @tkountis @idelpivnitskiy

Reply:
The Builder is also kind of strange taking constructor args for the routeStrategyFactory and codecs

Reply:
I'm agreed that we can simplify and keep only the `Builder`.

> The Builder is also kind of strange taking constructor args for the routeStrategyFactory and codecs

Agreed that it's a bit weird for the builder, however, we need the `strategyFactory` and `supportedMessageCodings` upfront, before users may invoke builder methods for specific routes. If we move `strategyFactory` and `supportedMessageCodings` as builder methods, users have to set them before invoking other builder methods, which we tried to avoid.

Reply:
⬆️

Reply:
No action required for this PR. 

First step would be to make the ServiceFactory(ServiceFactory.Builder) constructor public and deprecate the other ServiceFactory constructors in favor of using the ServiceFactory.Builder methods.